crates/ruff_python_formatter/src/other/with_item.rs

```diff
@@ -7,6 +9,16 @@ pub struct FormatWithItem;
 
 impl FormatNodeRule<WithItem> for FormatWithItem {
     fn fmt_fields(&self, item: &WithItem, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let WithItem {
+            range: _,
+            context_expr,
+            optional_vars,
+        } = item;
+        // parentheses added by parent StmtWith
+        write!(f, [context_expr.format().with_options(Parenthesize::Never)])?;
+        if let Some(optional_vars) = optional_vars {
+            write!(f, [space(), text("as"), space(), optional_vars.format()])?;
+        }
```

Comment:
Not sure where we should plate the `as ... name`. Should we place the `as` on a new line (as we do for binary operators) or keep it on the same line?

Reply:
black seems to keep on the same line i think